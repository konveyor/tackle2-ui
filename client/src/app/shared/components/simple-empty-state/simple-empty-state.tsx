import React from "react";
import {
  EmptyState,
  EmptyStateBody,
  EmptyStateIcon,
  EmptyStateVariant,
  Title,
} from "@patternfly/react-core";

export interface SimpleEmptyStateProps {
  icon?: any;
  title: string;
  description?: string;
  primaryAction?: any;
}

export const SimpleEmptyState: React.FC<SimpleEmptyStateProps> = ({
  icon,
  title,
  description,
  primaryAction,
}) => {
  return (
    <EmptyState variant={EmptyStateVariant.small}>
      {icon && <EmptyStateIcon icon={icon} />}
      <Title headingLevel="h2" size="lg">
        {title}
      </Title>
      {description && <EmptyStateBody>{description}</EmptyStateBody>}
      {primaryAction}
    </EmptyState>
  );
};
