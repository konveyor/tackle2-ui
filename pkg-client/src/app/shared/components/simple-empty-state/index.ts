export { SimpleEmptyState } from "./simple-empty-state";
