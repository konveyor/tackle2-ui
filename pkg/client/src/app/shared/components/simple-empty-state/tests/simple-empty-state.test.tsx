import React from "react";
import { shallow } from "enzyme";
import { AdIcon } from "@patternfly/react-icons/dist/esm/icons/ad-icon";

import { SimpleEmptyState } from "../simple-empty-state";

describe("SimpleEmptyState", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(<SimpleEmptyState title="my title" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders with icon", () => {
    const wrapper = shallow(
      <SimpleEmptyState title="my title" icon={AdIcon} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders with description", () => {
    const wrapper = shallow(
      <SimpleEmptyState title="my title" description="my description" />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders with primaryAction", () => {
    const wrapper = shallow(
      <SimpleEmptyState
        title="my title"
        description="my description"
        primaryAction={<button>My action</button>}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
