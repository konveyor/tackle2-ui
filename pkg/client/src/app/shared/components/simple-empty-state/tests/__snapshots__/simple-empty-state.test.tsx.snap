// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleEmptyState Renders with description 1`] = `
<EmptyState
  variant="small"
>
  <Title
    headingLevel="h2"
    size="lg"
  >
    my title
  </Title>
  <EmptyStateBody>
    my description
  </EmptyStateBody>
</EmptyState>
`;

exports[`SimpleEmptyState Renders with icon 1`] = `
<EmptyState
  variant="small"
>
  <Title
    headingLevel="h2"
    size="lg"
  >
    my title
  </Title>
</EmptyState>
`;

exports[`SimpleEmptyState Renders with primaryAction 1`] = `
<EmptyState
  variant="small"
>
  <Title
    headingLevel="h2"
    size="lg"
  >
    my title
  </Title>
  <EmptyStateBody>
    my description
  </EmptyStateBody>
  <button>
    My action
  </button>
</EmptyState>
`;

exports[`SimpleEmptyState Renders without crashing 1`] = `
<EmptyState
  variant="small"
>
  <Title
    headingLevel="h2"
    size="lg"
  >
    my title
  </Title>
</EmptyState>
`;
